model/host/db.go

```diff
@@ -130,6 +137,45 @@ func AllIdleEphemeral() ([]Host, error) {
 	return Find(query)
 }
 
+// IdleEphemeralGroupedByDistroId groups and collates the following grouped and ordered by {distro.Id: 1}:
+// - []host.Host of ephemeral hosts without containers which having no running task, ordered by {host.CreationTime: 1}
+// - the total number of ephemeral hosts with status: evergreen.HostRunning
+func IdleEphemeralGroupedByDistroId() ([]IdleHostsByDistroID, error) {
+	var idlehostsByDistroID []IdleHostsByDistroID
+	pipeline := []mgobson.M{
+		{
+			"$match": mgobson.M{
+				StartedByKey:     evergreen.User,
+				StatusKey:        evergreen.HostRunning,
+				ProviderKey:      mgobson.M{"$in": evergreen.ProviderSpawnable},
+				HasContainersKey: mgobson.M{"$ne": true},
+			},
+		},
+		{
+			"$sort": mgobson.M{CreateTimeKey: 1},
+		},
+		{
+			"$group": mgobson.M{
+				"_id":                             "$" + bsonutil.GetDottedKeyName(DistroKey, distro.IdKey),
+				HostsByDistroRunningHostsCountKey: mgobson.M{"$sum": 1},
+				HostsByDistroIdleHostsKey:         mgobson.M{"$push": bson.M{"$cond": []interface{}{mgobson.M{"$eq": []interface{}{"$running_task", mgobson.Undefined}}, "$$ROOT", mgobson.Undefined}}},
+			},
+		},
+		{
+			"$sort": mgobson.M{"_id": 1},
```

Comment:
I'm confused by this one. doesn't this undo the sort by create time (which is what we want more?)

Reply:
There is an existing test -`TestIdleEphemeralGroupedByDistroId` - in `model/host/host_test.go` to check that `distroID`s are ordered by `{"distro._id": 1}` and that their respective `IdleHosts` are sorted from oldest to newest `CreationTime`.

Reply:
Removed per our discussion